upup/pkg/fi/fitasks/keypair.go

```diff
@@ -268,3 +287,33 @@ func parsePkixName(s string) (*pkix.Name, error) {
 
 	return name, nil
 }
+
+func (e *Keypair) ensureResources() {
+	if e.certificate == nil {
+		e.certificate = &fi.TaskDependentResource{Task: e}
+	}
+	if e.certificateSHA1Fingerprint == nil {
+		e.certificateSHA1Fingerprint = &fi.TaskDependentResource{Task: e}
+	}
+}
+
+func (e *Keypair) setResources(cert *pki.Certificate) error {
+	e.ensureResources()
```

Comment:
I prefer to do stuff like this more lazily, not calculating the sha1 unless some previous ModelBuilder has called `CertificateSHA1Fingerprint()`

Reply:
Fair - as it is so cheap and to avoid the ordering requirement (must have called CertificateSHA1Fingerprint before setResources), is it OK to leave as is?

Reply:
It's cheap enough to leave as-is.